### PR TITLE
provide is_private method on GitProjects

### PR DESCRIPTION
**Pull request by dhodovsk**

### CODE REVIEW

File: ogr/services/pagure/project.py [hunk 1]
```diff
@@ -291,6 +291,19 @@ def exists(self):
...
+        Is this repo private? (accessible only by users with granted access)
+
+        :return: if yes, return True
+        """
+        if self.service.instance_url == "https://src.fedoraproject.org":
+            # private repositories are not allowed in fedora
+            return False
+        raise NotImplementedError(
+            f"is_private is not implemented for {self.service.instance_url}"
+        )
```
Comment (lachmanfrantisek) [inline ogr/services/pagure/project.py]: - A little hacky..;) => Can you please create an issue for the real implementation? (So we can discuss if it is possible to implement it.)
- I am not sure about the exact match of `"https://src.fedoraproject.org"`.
  - Will the instance URL always be with `https`? If you check that from packit, I am ok with merging as is. I am just not 100% sure...

File: tests/integration/test_github.py [hunk 2]
```diff
@@ -359,6 +359,9 @@ def test_get_web_url(self):
     def test_full_repo_name(self):
         assert self.ogr_project.full_repo_name == "packit-service/ogr"
 
+    def test_is_private(self):
+        assert not self.ogr_project.is_private()
```
Comment (lachmanfrantisek) [inline tests/integration/test_github.py]: Is it possible to test some private repository? Is it possible to create one in our organisation?

File: tests/integration/test_gitlab.py [hunk 3]
```diff
@@ -172,6 +172,9 @@ def test_pr_permissions(self):
         assert self.project.can_merge_pr("lachmanfrantisek")
         assert not self.project.can_merge_pr("not_existing_user")
 
+    def test_is_private(self):
+        assert not self.project.is_private()
```
Comment (lachmanfrantisek) [inline tests/integration/test_gitlab.py]: Same as above.

File: tests/integration/test_github.py (hunk 2 above)
Comment (dhodovsk) [reply tests/integration/test_github.py]: Our current plan does not support private repositories. We can create it on one of our users, but while re-generating the test data, the person would have to provide their own repository with their github key.

File: ogr/services/pagure/project.py (hunk 1 above)
Comment (dhodovsk) [reply ogr/services/pagure/project.py]: `https://src.fedoraproject.org` is the default and also used in packit service.
I agree that it's hacky, but I was not able to find good info in pagure doc for obtaining this info. But if it's important to you, I'll contact some human.

File: tests/integration/test_github.py (hunk 2 above)
Comment (lachmanfrantisek) [reply tests/integration/test_github.py]: OK, but we don't need to share the keys. Just create the repository with the same name and use that repository in tests. (The namespace should be the user of the token.)

File: tests/integration/test_gitlab.py (hunk 3 above)
Comment (lachmanfrantisek) [reply tests/integration/test_gitlab.py]: But in GitLab, we can create private repositories easily.

File: ogr/services/pagure/project.py (hunk 1 above)
Comment (dhodovsk) [reply ogr/services/pagure/project.py]: I tried to make it a bit nicer and talked with pingou - here is the issue: https://github.com/packit-service/ogr/issues/330

Comment (lachmanfrantisek) [reply ogr/services/pagure/project.py]: Thanks!

File: ogr/services/pagure/project.py [hunk 4]
```diff
@@ -291,6 +292,20 @@ def exists(self):
         response = self._call_project_api_raw()
         return response.ok
 
+    def is_private(self) -> bool:
+        """
+        Is this repo private? (accessible only by users with granted access)
+
+        :return: if yes, return True
+        """
+        host = urlparse(self.service.instance_url).hostname
```
Comment (lachmanfrantisek) [inline ogr/services/pagure/project.py]: We already have the parsing ability in `./parsing.py` that can be used instead, but  it's not so important.

```python
In [1]: from ogr.parsing import parse_git_repo                                        

In [2]: str(parse_git_repo("https://pagure.io"))                                      
Out[2]: "RepoUrl(repo='', namespace='None', is_fork=False, hostname='pagure.io', scheme='https')"
```

File: tests/integration/test_github.py [hunk 5]
```diff
@@ -359,6 +359,16 @@ def test_get_web_url(self):
     def test_full_repo_name(self):
         assert self.ogr_project.full_repo_name == "packit-service/ogr"
 
+    def test_is_not_private(self):
+        # when regenerating this test with your gitlab token, use your own private repository
+        private_project = self.service.get_project(
+            namespace="dhodovsk", repo="playground"
+        )
+        assert private_project.is_private()
```
Comment (lachmanfrantisek) [inline tests/integration/test_github.py]: Thanks!

**Comment by lachmanfrantisek** [comment]
>     * [ogr-pre-commit ](https://softwarefactory-project.io/zuul/t/local/build/76c744e8686e49b28b6cc569d969157b) : FAILURE in 3m 33s

Needs rebase...

**Comment by dhodovsk** [comment]
yup, I can read :)

**Comment by lachmanfrantisek** [comment]
> yup, I can read :)

Sorry.

**Pull request merged**